Makefile

```diff
@@ -48,35 +48,58 @@ word-hyphen = $(word $2,$(subst -, ,$1))
 
 .EXPORT_ALL_VARIABLES:
 
-.PHONY: build-aws-openzfs-csi-driver
-build-aws-openzfs-csi-driver:
-	mkdir -p bin
-	CGO_ENABLED=0 GOOS=linux go build -ldflags ${LDFLAGS} -o bin/aws-fsx-openzfs-csi-driver ./cmd/
+.PHONY: linux/$(ARCH) bin/aws-fsx-openzfs-csi-driver
+linux/$(ARCH): bin/aws-fsx-openzfs-csi-driver
+bin/aws-fsx-openzfs-csi-driver: | bin
+	CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) go build -mod=mod -ldflags ${LDFLAGS} -o bin/aws-fsx-openzfs-csi-driver ./cmd/
 
+# Builds all images
 .PHONY: all
 all: all-image-docker
 
+# Builds all images and pushes them
 .PHONY: all-push
-all-push:
-	docker buildx build \
-		--no-cache-filter=linux-amazon \
-		--platform=$(PLATFORM) \
-		--progress=plain \
-		--target=$(OS)-$(OSVERSION) \
-		--output=type=registry \
-		-t=$(IMAGE):$(TAG) \
-		.
-	touch $@
+all-push: push-manifest-and-images
+
+.PHONY: push-manifest-and-images
+push-manifest-and-images: create-manifest-and-images
```

Comment:
Why have a separate all-push if it just calls `push-manifest-and-images`. In the Lustre CSI Driver we removed `push-manifest-and-images`